ext/enchant/enchant.c

```diff
@@ -454,25 +485,14 @@ PHP_FUNCTION(enchant_broker_request_dict)
 		RETURN_FALSE;
 	}
 
-	d = enchant_broker_request_dict(pbroker->pbroker, (const char *)tag);
-	if (d) {
-		pos = pbroker->dictcnt++;
-		if (pbroker->dictcnt) {
-			pbroker->dict = (enchant_dict **)erealloc(pbroker->dict, sizeof(enchant_dict *) * pbroker->dictcnt);
-		} else {
-			pbroker->dict = (enchant_dict **)emalloc(sizeof(enchant_dict *));
-			pos = 0;
-		}
-
-		dict = pbroker->dict[pos] = (enchant_dict *)emalloc(sizeof(enchant_dict));
-		dict->id = pos;
-		dict->pbroker = pbroker;
-		dict->pdict = d;
-		pbroker->dict[pos] = dict;
+	pdict = enchant_broker_request_dict(pbroker->pbroker, (const char *)tag);
+	if (pdict) {
+		pbroker->nb_dict++;
 
-		dict->rsrc = zend_register_resource(dict, le_enchant_dict);
-		GC_ADDREF(pbroker->rsrc);
-		RETURN_RES(dict->rsrc);
+		object_init_ex(return_value, enchant_dict_ce);
+		dict = Z_ENCHANT_DICT_P(return_value);
+		dict->pdict =pdict;
```

Comment:
```suggestion
		dict->pdict = pdict;
```

Reply:
Fixed this (as well as adding an upgrading notes) when committing to master